### PR TITLE
More Invoicing Support

### PR DESCRIPTION
**Pull request by SkylerSeamans**
Support for Offline Payments and the Payment Transaction object.

### CODE REVIEW

**Comment by larsacus** [comment]
Looks good to me 👍

File: TeamSnapSDK/SDK/DataTypes/TSDKInvoice.m [hunk 1]
```diff
@@ -44,4 +45,25 @@ - (TSDKInvoiceStatus)invoiceStatus {
     }
 }
 
+- (void)makePayment:(NSDecimalNumber * _Nonnull)amount method:(TSDKInvoiceOfflinePaymentMethod)method note:(NSString * _Nullable)note completion:(TSDKSimpleCompletionBlock _Nullable)completion {
+    TSDKCollectionCommand *command;
+    switch (method) {
+        case TSDKInvoiceOfflinePaymentMethodCash:
+            command = [TSDKInvoicePayment commandForKey:@"pay_offline_cash"];
```
Comment (jrahaim) [inline TeamSnapSDK/SDK/DataTypes/TSDKInvoice.m:52]: These should be copies.

File: TeamSnapSDK/SDK/DataTypes/TSDKInvoice.m [hunk 2]
```diff
@@ -44,4 +45,25 @@ - (TSDKInvoiceStatus)invoiceStatus {
...
+            break;
+        case TSDKInvoiceOfflinePaymentMethodCheck:
+            command =[TSDKInvoicePayment commandForKey:@"pay_offline_cash"];
```
Comment (jrahaim) [inline TeamSnapSDK/SDK/DataTypes/TSDKInvoice.m:55]: These should be copies.

File: TeamSnapSDK/SDK/DataTypes/TSDKInvoicePaymentTransaction.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+//
+
+#import <Foundation/Foundation.h>
+#import "TSDKCollectionObject.h"
+#import "TSDKObjectsRequest.h"
+
+@interface TSDKInvoicePaymentTransaction : TSDKCollectionObject
+
+@property (nonatomic, weak, nullable) NSString * detail; //Example: **NULL**
+@property (nonatomic, assign) NSInteger processingFee; //Example: 0
```
Comment (jrahaim) [inline TeamSnapSDK/SDK/DataTypes/TSDKInvoicePaymentTransaction.h]: Money should be NSDecimalNumber

File: TeamSnapSDK/SDK/DataTypes/TSDKInvoicePaymentTransaction.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+@property (nonatomic, assign) NSInteger amountWithProcessingFee; //Example: 25
+@property (nonatomic, weak, nullable) NSString * amountWithCurrency; //Example: $25.00
+@property (nonatomic, weak, nullable) NSString * paymentType; //Example: Cash
+@property (nonatomic, assign) NSInteger amount; //Example: 25
```
Comment (jrahaim) [inline TeamSnapSDK/SDK/DataTypes/TSDKInvoicePaymentTransaction.h]: NSDecimalNumber

File: TeamSnapSDK/SDK/DataTypes/TSDKInvoicePaymentTransaction.h [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+#import <Foundation/Foundation.h>
+#import "TSDKCollectionObject.h"
+#import "TSDKObjectsRequest.h"
+
+@interface TSDKInvoicePaymentTransaction : TSDKCollectionObject
+
+@property (nonatomic, weak, nullable) NSString * detail; //Example: **NULL**
+@property (nonatomic, assign) NSInteger processingFee; //Example: 0
+@property (nonatomic, assign) NSInteger amountWithProcessingFee; //Example: 25
```
Comment (jrahaim) [inline TeamSnapSDK/SDK/DataTypes/TSDKInvoicePaymentTransaction.h]: NSDecimalNumber

**Review by jrahaim — CHANGES_REQUESTED** [review]

File: TeamSnapSDK/SDK/DataTypes/TSDKInvoice.m (hunk 1 above)
Comment (jrahaim) [reply TeamSnapSDK/SDK/DataTypes/TSDKInvoice.m:52]: I stand corrected, it returns a copy. Sorry.

File: TeamSnapSDK/SDK/DataTypes/TSDKInvoice.m (hunk 2 above)
Comment (jrahaim) [reply TeamSnapSDK/SDK/DataTypes/TSDKInvoice.m:55]: I stand corrected, it returns a copy. Sorry.

**Comment by jrahaim** [comment]
Wont copile for tests. 
`/Users/jason/gitSource/TeamSnapSDK/TeamSnapSDK/TeamSnapSDK.h:104:9: 'TeamSnapSDK/TSDKInvoicePayment.h' file not found`

I had the same issue when I tries to  add types to the array parameters on my completion blocks which is why they were left out.

**Comment by SkylerSeamans** [comment]
All changes addressed.

The tests issue is because the new files target membership was set to "project" instead of "public"

**Review by jrahaim — APPROVED** [review]

**Comment by jrahaim** [comment]
:+1: Unit test
:+1: Code.

**Pull request merged**